tedana/utils.py

```diff
@@ -49,37 +49,58 @@ def reshape_niimg(data):
     return fdata
 
 
-def make_adaptive_mask(data, mask=None, getsum=False, threshold=1):
-    """
-    Make map of `data` specifying longest echo a voxel can be sampled with.
+def make_adaptive_mask(data, mask=None, threshold=1):
+    """Make map of `data` specifying longest echo a voxel can be sampled with.
 
     Parameters
     ----------
     data : (S x E x T) array_like
-        Multi-echo data array, where `S` is samples, `E` is echos, and `T` is
-        time
+        Multi-echo data array, where `S` is samples, `E` is echos, and `T` is time.
     mask : :obj:`str` or img_like, optional
         Binary mask for voxels to consider in TE Dependent ANAlysis. Default is
         to generate mask from data with good signal across echoes
-    getsum : :obj:`bool`, optional
-        Return `masksum` in addition to `mask`. Default: False
     threshold : :obj:`int`, optional
         Minimum echo count to retain in the mask. Default is 1, which is
         equivalent not thresholding.
 
     Returns
     -------
     mask : (S,) :obj:`numpy.ndarray`
-        Boolean array of voxels that have sufficient signal in at least one
-        echo
+        Boolean array of voxels that have sufficient signal in at least ``threshold`` echos.
     masksum : (S,) :obj:`numpy.ndarray`
-        Valued array indicating the number of echos with sufficient signal in a
-        given voxel. Only returned if `getsum = True`
+        Valued array indicating the number of echos with sufficient signal in a given voxel.
+
+    Notes
+    -----
+    The adaptive mask is constructed from two methods:
+
+    1.  Count the total number of echoes in each voxel that have "good" data.
+
+        a.  Calculate the 33rd percentile of values in the first echo,
+            based on voxel-wise mean over time.
+        b.  Identify voxels where the first echo's mean value is equal to the 33rd percentile.
+            Basically, this identifies "exemplar" voxels reflecting the 33rd percentile.
+
+            -   The 33rd percentile is arbitrary.
+        c.  Calculate 1/3 of the mean value of the exemplar voxels for each echo.
+
+            -   This is the threshold for "good" data.
+            -   The 1/3 value is arbitrary.
+        d.  Only retain the highest value for each echo, across exemplar voxels.
+        e.  For each voxel, count the number of echoes that have a mean value greater than the
+            corresponding echo's threshold.
```

Comment:
```suggestion
        a.  Calculate the 33rd percentile of values in the first echo,
            based on voxel-wise mean over time.
        b.  Identify the voxel where the first echo's mean value is equal to the 33rd percentile.
            Basically, this identifies "exemplar" voxel reflecting the 33rd percentile.

            -   The 33rd percentile is arbitrary.
            -   If more than one voxel equals the 33rd percentile, keep all of them.
        c.  Calculate 1/3 of the mean value of the exemplar voxel for each echo.

            -   This is the threshold for "good" data.
            -   The 1/3 value is arbitrary.
        d.  If there was more than one exemplar voxel, retain the the highest value for each echo.
        e.  For each voxel, count the number of echoes that have a mean value greater than the
            corresponding echo's threshold.
```

Reply:
The diff here isn't clear. Could you summarize what changes you made?

Also, I'd like to expand on the logic behind this method. Does this sound right?

> This method assumes that exemplar voxels' signal at later echoes are also reasonable, and that any voxels whose values at a given echo are less than 1/3 of the exemplar voxels' values at that echo are affected by dropout.

Reply:
I should also probably make a note in the documentation that this procedure depends on the voxels included in the base mask- if your brain mask is bad, then it will impact your adaptive mask.

Reply:
I mostly changed the 33rd percentiles voxel**S** to voxel and then added a bullet that, if more than one voxel is at exactly the 33rd percentile then keep all of them and pick the maximum of each echo below. Content-wise I think your original is correct, but I tried to clarify.

Reply:
Ah, thank you for the clarification. I will accept the change and add my summary of the logic.